### PR TITLE
Update to Flux 2.1.2

### PR DESCRIPTION
**Pull request by tamalsaha**
#### What this PR does / why we need it:
Update to Flux 2.1.2

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] helm-docs are updated
- [x] Helm chart is tested
- [x] Update artifacthub.io/changes in Chart.yaml
- [x] Run `make reviewable`

### CODE REVIEW

**Comment by tamalsaha** [comment]
@stefanprodan , I have updated the snap files. It seems the updated unittest uses different indentation

```
> helm unittest --file 'tests/*.yaml' charts/flux2/ -u
> helm unittest --file 'tests/*.yaml' charts/flux2-multi-tenancy/ -u
> helm unittest --file 'tests/*.yaml' charts/flux2-notification/ -u
```

Also, fixed one test since `isNull` and `isEmpty` has different meaning now.
https://github.com/helm-unittest/helm-unittest/issues/134#issuecomment-1512099216

File: charts/flux2/Chart.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 annotations:
   artifacthub.io/changes: |
-    - "feat: enable labeling of deployments"
+    - "feat: upgrade to FluxCD 2.1.2"
 apiVersion: v2
 appVersion: 2.1.1
```
Comment (stefanprodan) [inline charts/flux2/Chart.yaml:5]: This needs to be bumped to 2.1.2

**Comment by tamalsaha** [comment]
I updated the variable `FLUX2_VERSION ?= v2.1.2` in Makefile. Then `make reviewable` make most of the rest of the changes.

PTAL.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @tamalsaha 🏅

**Pull request merged**